WcaOnRails/app/models/competition.rb

```diff
@@ -115,7 +116,11 @@ def create_id_and_cell_name
         # By replacing accented chars with their ascii equivalents, and then
         # removing everything that isn't a digit or a character.
         safe_name_without_year = ActiveSupport::Inflector.transliterate(name_without_year).gsub(/[^a-z0-9]+/i, '')
-        self.id = safe_name_without_year[0...(MAX_ID_LENGTH - year.length)] + year
+        self.id = safe_name_without_year[0...(MAX_FRIENDLY_ID_LENGTH - year.length)] + year
+      end
+      if friendly_id.blank?
+        safe_name_without_year = ActiveSupport::Inflector.transliterate(name_without_year).gsub(/[^a-z0-9]+/i, '')
```

Comment:
Did you consider moving `safe_name_without_year` up outside of these `if` statements instead of copying it? I don't think performance is a real concern here.